examples/paywall-tester/src/main/java/com/revenuecat/paywallstester/ui/screens/paywall/PaywallScreenViewModel.kt

```diff
@@ -0,0 +1,56 @@
+package com.revenuecat.paywallstester.ui.screens.paywall
+
+import android.app.Application
+import androidx.lifecycle.AndroidViewModel
+import androidx.lifecycle.SavedStateHandle
+import com.revenuecat.purchases.Purchases
+import com.revenuecat.purchases.getOfferingsWith
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.update
+
+interface PaywallScreenViewModel {
+    companion object {
+        const val OFFERING_ID_KEY = "offering_id"
+    }
+    val state: StateFlow<PaywallScreenState>
+}
+
+class PaywallScreenViewModelImpl(
+    application: Application,
+    savedStateHandle: SavedStateHandle,
+) : AndroidViewModel(application), PaywallScreenViewModel {
+
+    override val state: StateFlow<PaywallScreenState>
+        get() = _state.asStateFlow()
+    private val _state: MutableStateFlow<PaywallScreenState> = MutableStateFlow(PaywallScreenState.Loading)
+
+    private val offeringId = savedStateHandle.get<String?>(PaywallScreenViewModel.OFFERING_ID_KEY)
+
+    init {
+        updateOffering()
```

Comment:
We need to refetch the offering, since after navigating to this screen, we only have the offeringId.

Reply:
We are also loading in the `PaywallViewModel` right? Do we need to do it in both places?

Reply:
Got it! Passing an offering and getting the current if null makes sense!